File: c++/task3.cpp
#include <iostream>
#include <vector>
using namespace std;

int digitSum(int n) {
    int sum = 0;
    while (n) {
        sum += n % 10;
        n /= 10;
    }
    return sum;
}

int digitProduct(int n) {
    int product = 1;
    while (n) {
        product *= n % 10;
        n /= 10;
    }
    return product;
}

int main() {
    setlocale(LC_ALL, "ru_RU.UTF-8");

    int size;
    cout << "Сколько чисел вы хотите ввести? ";
    cin >> size;

    vector<int> numbers(size);
    cout << "Введите " << size << " чисел через пробел: ";
    for (int i = 0; i < size; ++i) cin >> numbers[i];

    cout << "Позиции чисел, где сумма цифр меньше произведения (нумерация с 1): ";
    for (int i = 0; i < size; ++i) {
        if (digitSum(numbers[i]) < digitProduct(numbers[i])) {
            cout << i + 1 << " ";
        }
    }
    cout << endl;

    return 0;
}

File: go/task3.go
package main

import (
	"fmt"
)

func digitSum(n int) int {
	sum := 0
	for n != 0 {
		sum += n % 10
		n /= 10
	}
	return sum
}

func digitProduct(n int) int {
	product := 1
	for n != 0 {
		product *= n % 10
		n /= 10
	}
	return product
}

func main() {
	var size int
	fmt.Print("Сколько чисел вы хотите ввести? ")
	fmt.Scan(&size)

	numbers := make([]int, size)
	fmt.Println("Введите", size, "чисел через пробел:")
	for i := 0; i < size; i++ {
		fmt.Scan(&numbers[i])
	}

	fmt.Print("Позиции, где сумма цифр < произведения (нумерация с 1): ")
	for i, n := range numbers {
		if digitSum(n) < digitProduct(n) {
			fmt.Printf("%d ", i+1)
		}
	}
	fmt.Println()
}
